spark3/src/main/scala/org/apache/spark/sql/catalyst/utils/PlanUtils.scala

```diff
@@ -61,7 +61,8 @@ object PlanUtils {
     if (Spark3VersionUtil.isSpark30) {
       repartitionByExpressionCtor.newInstance(partitionExpressions, child, Integer.valueOf(numPartitions))
     } else {
-      repartitionByExpressionCtor.newInstance(partitionExpressions, child, Some(numPartitions))
+      // SPARK-32056: Coalesce partitions for repartition by expressions when AQE is enabled
```

Comment:
This comment isn't very helpful, it just describes a Spark issue. Could you remove it or update it to something that explains a part of the next line? Something like "Do not pass numPartitions because it is set automatically for AQE"